File: src/com/jda/functionalprograms/ZeroSumTriples.java
package com.jda.functionalprograms;

import com.jda.utility.Reader;
import com.jda.utility.Utility;

public
class ZeroSumTriples {

  public static
  void main(String[] args) {
    Reader reader = new Reader();
    Utility.printLine("How many integers do you want to read?");
    int n = reader.readInteger();
    Utility.printLine("Enter " + n + " integers");
    int[] array = reader.readIntegerArray(n);
    int count = findZeroSumTriples(array);
    Utility.printLine("There are " + count + " zero sum triples");
  }

  private static
  int findZeroSumTriples(int[] array) {
    int n = array.length;
    int count = 0;
    for (int i = 0; i < n - 2; i++) {
      for (int j = i + 1; j < n - 1; j++) {
        for (int k = j + 1; k < n; k++) {
          if (array[i] + array[j] + array[k] == 0) {
            count++;
            System.out.println(array[i] + " " + array[j] + " " + array[k]);
          }
        }
      }
    }
    return count;
  }
}

File: src/com/jda/functionalprograms/ZeroSumTriples.kt
package com.jda.functionalprograms

import com.jda.utility.Utility

object ZeroSumTriples {

    @JvmStatic
    fun main(args: Array<String>) {
        val reader = Utility().Reader()
        Utility.printLine("How many integers do you want to read?")
        val n = reader.readInteger()
        Utility.printLine("Enter $n integers")
        val array = reader.readIntegerArray(n)
        val count = findZeroSumTriples(array)
        Utility.printLine("There are $count zero sum triples")
    }

    private fun findZeroSumTriples(array: IntArray): Int {
        val n = array.size
        var count = 0
        for (i in 0 until n - 2) {
            for (j in i + 1 until n - 1) {
                for (k in j + 1 until n) {
                    if (array[i] + array[j] + array[k] == 0) {
                        count++
                        println(array[i].toString() + " " + array[j] + " " + array[k])
                    }
                }
            }
        }
        return count
    }
}
